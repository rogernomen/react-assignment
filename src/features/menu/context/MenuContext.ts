import React from "react";

interface MenuContext {
	activeMenuIndex: number;
	setActiveMenuIndex: React.Dispatch<React.SetStateAction<number>>;
}

const initialMenuContext: MenuContext = {
	activeMenuIndex: 0,
	// eslint-disable-next-line @typescript-eslint/no-empty-function
	setActiveMenuIndex: () => {}
};

const MenuContext = React.createContext(initialMenuContext);
export const MenuContextProvider = MenuContext.Provider;
export default MenuContext;
