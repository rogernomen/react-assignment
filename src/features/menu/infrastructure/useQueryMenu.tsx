import { useQuery, UseQueryResult } from "react-query";
import { Menu } from "../../../../bff/dist/models/Menu";
import api from "../../../infrastructure/api";

export const useQueryMenu = (): UseQueryResult<Menu, Error> => {
	return useQuery<Menu, Error>(
		["fetch-menu"],
		() => {
			return api.get("menu");
		},
		{ keepPreviousData: true }
	);
};
