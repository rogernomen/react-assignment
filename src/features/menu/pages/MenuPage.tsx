import React, { useState } from "react";
import { useQueryMenu } from "../infrastructure/useQueryMenu";
import MenuNav from "../components/MenuNav";
import MenuContent from "../components/MenuContent";
import MenuContext from "../context/MenuContext";

const MenuPage: React.FC = (): null | React.ReactElement => {
	const { data: menu } = useQueryMenu();
	const [activeMenuIndex, setActiveMenuIndex] = useState<number>(0);
	const menuContextProviderValue = { activeMenuIndex, setActiveMenuIndex };

	return menu ? (
		<MenuContext.Provider value={menuContextProviderValue}>
			<MenuNav menuSections={menu.menuSections} />
			<MenuContent menuItems={menu.menuSections[activeMenuIndex].menuItems} />
		</MenuContext.Provider>
	) : null;
};

export default MenuPage;
