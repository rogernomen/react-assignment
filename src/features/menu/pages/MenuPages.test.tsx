import React from "react";
import renderer from "react-test-renderer";
import MenuPage from "./MenuPage";
import MenuContext from "../context/MenuContext";
import { useQueryMenu } from "../infrastructure/useQueryMenu";

const mockedUseQueryMenu = useQueryMenu as jest.Mock<any>;
jest.mock("../infrastructure/useQueryMenu");

describe("the MenuPage Component", () => {
	it("should be rendered", () => {
		expect.assertions(1);
		mockedUseQueryMenu.mockImplementation(() => ({ isLoading: true }));
		const contextValue = {
			activeMenuIndex: 0,
			setActiveMenuIndex: () => jest.fn()
		};

		const component = renderer.create(
			<MenuContext.Provider value={contextValue}>
				<MenuPage></MenuPage>
			</MenuContext.Provider>
		);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
