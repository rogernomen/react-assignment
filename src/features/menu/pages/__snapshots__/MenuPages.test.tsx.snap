// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the MenuPage Component should be rendered 1`] = `null`;
